packages/runtime/container-runtime/api-report/container-runtime.api.md

```diff
@@ -442,6 +442,7 @@ export interface IGCRuntimeOptions {
     gcAllowed?: boolean;
     runFullGC?: boolean;
     sessionExpiryTimeoutMs?: number;
+    tombstoneSweepDelayMs?: number;
```

Comment:
> Rajat: Trying to understand what our strategy is for the reaction time. For instance, If we see Tombstone errors in our Dashboard or via Incidents, how do we selectively reset sweep readiness of those specific documents? What tools do we have to specifically identify such documents and reset to save from sweep remotely?

Mark: It's up to the apps to use our available configs however their deployment supports. Auto-protection (what you describe) is my 2nd suggested future investment here.

Reply:
True. But for Loop scenarios specifically I am trying to understand how will this additional delay in Sweep help save those incorrectly Tombstoned documents?
> For a document being actively used that hits a GC-impacting bug, **_this will cause the bug to manifest as a Tombstone error first, giving some reaction time (operationally) before_** the data is permanently lost.

Reply:
I think they're planning to use something like 10-15 days.  So for a document that is actively being used, it will likely hit the Tombstone error in that window, they'll see the telemetry, and have some time to figure out what to do.  This is what their TSG will cover.

Reply:
Agreed, it's something Loop's TSG will cover, but we should make sure that details of that process (of saving incorrectly Tombstoned documents in the 10-15 days of extra time) are thought through. I am not clear on...
How and when will the app get the Document ID of incorrectly-Tombstoned document and how will the app disable sweep for it remotely? Will it be a proactive effort that everyday, however many incorrectly_Tombstoned log show up, app will go and scan all documents and turn off Sweep for them? Will we ever find ourselves in a situation where Sweep has to be turned off for all documents, because identifying negatively-impacted documents is not possible? 

Knowing these answer will help us understand that Two-Stage sweep is going to fully serve the purpose it is aiming to serve. fyi: @agarwal-navin

Reply:
There is a follow up task to do just that - Recovery mechanims between tombstone and sweep. I am not sure if this is the right time or place to discuss this. Sure, we should evaluate if doing this is worth but I thought we already established that. Maybe we can talk offline about this.